Ensuring transient value is an array

gedex opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR fixes a critical PHP 8.4 compatibility issue where the WooCommerce Helper updater crashes with a fatal `TypeError` when the `_woocommerce_helper_updates` transient contains malformed string data instead of the expected array.

**Problem:**
- On PHP 8.4, attempting to access array offsets on a string value throws a fatal `TypeError`
- The `WC_Helper_Updater::_update_check()` method was attempting to access `$data['hash']` and `$data['products']` without validating that `$data` is actually an array
- When the transient becomes corrupted or contains unexpected string data, fatal error is thrown

**Solution:**
- Add type validation to ensure `$data` is an array before attempting array access
- Changed condition from `if ( false !== $data )` to `if ( false !== $data && is_array( $data ) )`
- This allows the updater to safely handle malformed transient data by treating it as a cache miss and refreshing the data

Closes #61981
Closes WOOPLUG-5866

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**Testing the fix:**

1. Set up a test site with PHP 8.4 and WooCommerce 10.3.5+
2. Manually corrupt the transient to simulate the issue:
   ```
   pnpm wp-env run cli wp transient set _woocommerce_helper_updates ''
   ```

3. Visit WooCommerce Extensions page at http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions
4. Verify that the site loads without errors
5. Check that the helper updates functionality continues to work normally

**Expected behavior:**

No fatal errors occur

### Testing that has already taken place:

- Code review and analysis of the crash scenario from merchant logs (site 10488304-zen)
- Verification that the fix handles the specific PHP 8.4 TypeError mentioned in issue #61981
- Confirmed that adding `is_array()` check prevents array access on string values
- The fix is minimal and defensive, only adding type validation without changing existing logic

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message

Fix PHP 8.4 fatal error in WooCommerce Helper updater when transient contains malformed data

</details>

gedex committed:
Ensure transient is an array

Malformed value for _woocommerce_helper_updates transient can throw
Uncaught TypeError: Cannot access offset of type string on string.

Fixes #61981
Fixes WOOPLUG-5866

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

KokkieH reviewed (approved):
Confirmed I see the error on `trunk`. Page loads without errors on this branch.

gedex committed:
Add unit test for WC_Helper_Updater

gedex committed:
Fix missing member variable doc comment

mcliwanow commented on plugins/woocommerce/includes/admin/helper/class-wc-helper-updater.php:
```diff
@@ -619,7 +619,7 @@ private static function _update_check( $payload ) {
 		$data      = get_transient( $cache_key );
-		if ( false !== $data ) {
+		if ( false !== $data && is_array( $data ) ) {
 			if ( hash_equals( $hash, $data['hash'] ) ) {
```
For an abundance of caution, should we also check if `$data['products']` is an array?

gedex replied:
Sounds reasonable to prevent potential fatal error. We should also verify `$data['hash']`.

gedex replied:
Added helper method to check (with additional test cases) in 6fbf360500c2efc4b165e26be60eb430abe47d16.

gedex committed:
Add defensive validation helper for cached update data

Introduces `should_use_cached_update_data()` method to encapsulate and
strengthen validation of cached transient data. This provides more
comprehensive protection against malformed data that could cause fatal
errors.
